contracts/Collateralizer.sol

```diff
@@ -81,44 +78,35 @@ contract Collateralized is TermsContract {
     }
 
     /**
-     * Abstract interface for a function that returns the last timestamp
-     * at which any repayment is expected -- i.e., the debt agreement's
-     * term's ending.
+     * Transfers collateral from the debtor to the current contract, as custodian.
      *
      * @param agreementId bytes32 The debt agreement's ID
-     * @return _timestamp uint
-     */
-    function getTermEndTimestamp(
-        bytes32 agreementId
-    ) public view returns (uint _timestamp);
-
-    /**
-     * Hook that is called by the DebtKernel upon a debt agreement's
-     * issuance.  Parses out collateralization-specific parameters
-     * from the debt agreement's terms contract parameters and
-     * pulls collateral from the debtor based on those parameters.
-     *
-     * @param agreementId bytes32 The debt's agreement ID
-     * @param collateralizer address The debt agreement's collateralizer
-     * @return _success bool
+     * @param collateralizer address The owner of the asset being collateralized
      */
-    function registerTermStart(bytes32 agreementId, address collateralizer)
-        public
-        returns (bool _success)
+    function collateralize(
+        bytes32 agreementId,
+        address collateralizer
+    ) public returns (bool _success)
     {
-        require(msg.sender == debtKernelAddress);
-
-        // Fetch all relevant collateralization parameters
+        // require(collateralizationPermissions.isAuthorized(msg.sender));
```

Comment:
I'm not actually sure how to implement this @nadavhollander if you could give some guidance?